### PR TITLE
Bugfix/#81 editor spans and auto apply

### PR DESCRIPTION
**Pull request by mattcowan**
This pull request introduces robust handling for nested typography spans and multi-node selections in the typography stylist plugin, as well as comprehensive test coverage for these features. The changes significantly improve the plugin's ability to layer multiple typography properties, enforce nesting depth limits, and accurately parse inline styles at the cursor position.

### Nested Span Handling & Edge Case Management

* Added support for nested `<span class="typost-styled">` elements to layer multiple typography properties, with a maximum enforced nesting depth of 3 levels via the `validateNestingDepth()` utility function. Edge cases such as partial selections, feature spans nested within font property spans, and automatic cleanup of empty spans are handled. (`CLAUDE.md`)

### Multi-Node Selection & Fallback Utilities

* Implemented and thoroughly tested the `applyStylingSafeStringMethod` utility to handle multi-node selections while preserving nested span structure. The tests ensure correct handling of double/triple-nested spans, broad selections, attribute preservation, and edge cases such as null/undefined attributes. (`applyStylingSafeStringMethod.test.js`)

* Added and tested the `validateNestingDepth` utility to enforce and verify the maximum nesting depth, including graceful handling of invalid or non-typost-styled elements. (`applyStylingSafeStringMethod.test.js`)

### Inline Style Parsing Improvements

* Introduced comprehensive tests for `parseInlineStylesAtCursor`, covering detection of all inline style properties at the cursor position, inheritance rules for nested spans, boundary conditions, and backward compatibility with style attributes. (`parseInlineStylesAtCursor.test.js`)

### CODE REVIEW

File: tests/e2e/typography-stylist.spec.js [hunk 1]
```diff
@@ -476,4 +476,490 @@ test.describe('Typography Stylist - Inline Features', () => {
...
+    }
+    await page.keyboard.up('Shift');
+    await page.waitForTimeout(500);
+
+    await toggleButton.click();
+    await page.waitForTimeout(1000);
+
+    // Verify font-weight dropdown shows 700 (not "inherit")
+    const fontWeightValue = await fontWeightSelect.inputValue();
+    expect(fontWeightValue).toBe('700');
```
[the remark this answers is not in the document]
Comment (mattcowan) [reply tests/e2e/typography-stylist.spec.js:963]: [b98a362](https://github.com/mattcowan/typography-stylist-for-wordpress/pull/98/commits/b98a36201afcf47ccf4a1674015d6fc3836e9996)
